src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/MessageBoxInProcess.cs

```diff
@@ -0,0 +1,228 @@
+            var hr = VSConstants.E_NOTIMPL;
+
+            if (guidService == typeof(IVsMessageBoxService).GUID && riid == typeof(IVsMessageBoxService).GUID)
+            {
+                var comInterface = Marshal.GetComInterfaceForObject(this, typeof(IVsMessageBoxService));
+                if (comInterface != IntPtr.Zero)
+                {
+                    hr = Marshal.QueryInterface(comInterface, ref riid, out ppvObject);
+                    Marshal.Release(comInterface);
+                }
+            }
+
+            return hr;
+        }
+
+        public int ShowMessageBox(
+            IntPtr hWndOwner,
+            IntPtr hInstance,
+            string lpszText,
+            string lpszCaption,
+            uint dwStyle,
+            IntPtr lpszIcon,
+            IntPtr dwContextHelpId,
+            IntPtr pfnMessageBoxCallback,
+            uint dwLangID,
+            out int pidButton)
+        {
+            foreach (var handler in s_handlers)
+            {
+                var result = handler.Handle(lpszText, lpszCaption);
+                if (result != DialogResult.None)
+                {
+                    pidButton = (int)result;
+                    return VSConstants.S_OK;
+                }
+            }
+
+            Assert.True(
+                false,
+                $"""
+                Unexpected dialog box appeared.
+                Text: {lpszText}
+                Caption: {lpszCaption}
+                """);
+            throw ExceptionUtilities.Unreachable();
+        }
+    }
+
+    private sealed class HotReloadUIService : IOleServiceProvider, IHotReloadUIService
+    {
+        public int QueryService(ref Guid guidService, ref Guid riid, out IntPtr ppvObject)
+        {
+            ppvObject = IntPtr.Zero;
+            var hr = VSConstants.E_NOTIMPL;
+
+            if (guidService == typeof(IHotReloadUIService).GUID && riid == typeof(IHotReloadUIService).GUID)
+            {
+                var comInterface = Marshal.GetComInterfaceForObject(this, typeof(IHotReloadUIService));
+                if (comInterface != IntPtr.Zero)
+                {
+                    hr = Marshal.QueryInterface(comInterface, ref riid, out ppvObject);
+                    Marshal.Release(comInterface);
+                }
+            }
+
+            return hr;
+        }
+
+        public int ShowHotReloadDialog(bool isManaged, EditAndContinueResult result, string errorMessage)
+        {
+            foreach (var handler in s_hotReloadHandlers)
+            {
+                var hotReloadAction = handler.Handle(isManaged, result, errorMessage);
+                switch (hotReloadAction)
+                {
+                    case HotReloadAction.None:
+                        continue;
+
+                    case HotReloadAction.Cancel:
+                        goto case HotReloadAction.Edit;
+
+                    case HotReloadAction.Edit:
+                        //if (!isManaged)
```

Comment:
Do we want to keep these commentted codes?

Reply:
Also there are commented code in other cases.

Reply:
➡️ Yes (keep), mainly because if we ever need these, the text in the comment gives something to search for the reference implementation.